app/src/main/java/org/jellyfin/mobile/bridge/ExternalPlayer.kt

```diff
@@ -0,0 +1,148 @@
+package org.jellyfin.mobile.bridge
+
+import android.app.Activity
+import android.content.Intent
+import android.webkit.JavascriptInterface
+import org.jellyfin.mobile.MainActivity
+import org.jellyfin.mobile.R
+import org.jellyfin.mobile.player.ExoPlayerFormats
+import org.jellyfin.mobile.player.source.JellyfinMediaSource
+import org.jellyfin.mobile.utils.Constants
+import org.json.JSONException
+import org.json.JSONObject
+import org.koin.core.KoinComponent
+import timber.log.Timber
+
+class ExternalPlayer(private val activity: MainActivity) : KoinComponent {
+
+    private var mediaSource: JellyfinMediaSource?= null
+    private var playerIntent: Intent?= null
+
+    @JavascriptInterface
+    fun getSupportedFormats() = ExoPlayerFormats.supportedCodecs.toJSONString()
+
+    @JavascriptInterface
+    fun loadPlayer(args: String) {
+        try {
+            mediaSource = JellyfinMediaSource(JSONObject( args));
+            if(mediaSource?.playMethod.equals("DirectStream")) {
+                playerIntent = Intent(Intent.ACTION_VIEW).apply {
+                    setDataAndType(mediaSource?.uri, mediaSource?.mimeType)
+                    putExtra("title", mediaSource?.title)
+                    putExtra("position", mediaSource?.mediaStartMs)
+                    putExtra("return_result", true)
+                    putExtra("secure_uri", true)
+                }
+                activity.startActivityForResult(playerIntent, Constants.HANDLE_EXTERNAL_PLAYER)
+                Timber.d("Starting playback [id: ${mediaSource?.id}, title: ${mediaSource?.title}, playMethod: ${mediaSource?.playMethod}, mediaStartMs: ${mediaSource?.mediaStartMs}, mediaDurationMs: ${mediaSource?.mediaDurationMs}]")
+            } else {
+                Timber.d("Play Method '${mediaSource?.playMethod}' not tested, ignoring...")
+                notifyEvent(Constants.EVENT_CANCELED, "'${activity.getString(R.string.external_player_invalid_play_method)}'")
+            }
+        } catch (e: JSONException) {
+            Timber.e(e)
+        }
+    }
+
+    companion object {
+        private var mainActivity: MainActivity ?= null
```

Comment:
I'd suggest to pass this activity reference to the handle functions directly, to ensure that we never leak this context if in some case it isn't set to null again.

Reply:
I changed the way to call the functions, in theory there should be no problem with context leaks now.